nemo/collections/asr/models/ctc_bpe_models.py

```diff
@@ -319,7 +323,9 @@ def _setup_transcribe_dataloader(self, config: Dict) -> 'torch.utils.data.DataLo
         temporary_datalayer = self._setup_dataloader_from_config(config=DictConfig(dl_config))
         return temporary_datalayer
 
-    def change_vocabulary(self, new_tokenizer_dir: str, new_tokenizer_type: str):
+    def change_vocabulary(
+        self, new_tokenizer_dir: str, new_tokenizer_type: str, new_tokenizer_cfg: Optional[DictConfig] = None
```

Comment:
Hmm this isn't a nice API - you normally change the config by using fbe dir and type. 

Why not make the new argument overloaded - check if it's a str - assume it's a path, if it's a dict of a dictconfig, assume it's a new config and anything else raise an error. That way the api remains consistent.

Reply:
You can also assert that if config is given type must be agg and vice versa - if agg is given as type the first arv must be a dict or dictconfig

Reply:
Sorry do you mean change new_tokenizer_cfg to be either a string or a dictconfig and remove new_tokenizer dir?

Reply:
made new_tokenizer_dir a Union[str, DictConfig] for now